libpysal/weights/util.py

```diff
@@ -1552,23 +1566,36 @@ def fuzzy_contiguity(gdf, tolerance=0.005, buffering=False, drop=True, buffer=No
         if not buffer:
             # buffer each shape
             minx, miny, maxx, maxy = gdf.total_bounds
-            buffer = tolerance * 0.5 * abs(min(maxx-minx, maxy-miny))
+            buffer = tolerance * 0.5 * abs(min(maxx - minx, maxy - miny))
         # create new geometry column
         new_geometry = gdf.geometry.buffer(buffer)
         gdf['_buffer'] = new_geometry
         old_geometry_name = gdf.geometry.name
         gdf.set_geometry('_buffer', inplace=True)
-    assert gdf.sindex, 'GeoDataFrame must have a spatial index. Please make sure you have `libspatialindex` installed'
-    tree = gdf.sindex
+
     neighbors = {}
-    n,k = gdf.shape
-    for i in range(n):
-        geom = gdf.geometry.iloc[i]
-        hits = list(tree.intersection(geom.bounds))
-        possible = gdf.iloc[hits]
-        ids = possible.intersects(geom).index.tolist()
-        ids.remove(i)
-        neighbors[i] = ids
+    if GPD_08:
+        # query tree based on set predicate
+        inp, res = gdf.sindex.query_bulk(gdf.geometry, predicate=predicate)
+        # remove self hits
+        itself = inp == res
+        inp = inp[~itself]
+        res = res[~itself]
+
+        # extract index values of neighbors
+        for i, ix in enumerate(gdf.index):
+            ids = gdf.iloc[res[inp == i]].index.tolist()
```

Comment:
```suggestion
            ids = gdf.index[res[inp == i]].tolist()
```

If you only need the indices, instead of indexing the dataframe, you can directly index the index, which will be faster (and avoiding an unnecessary copy of the subset of the data)